modules/flakePartsModules.nix

```
@@ -0,0 +1,27 @@
+{ config, self, lib, flake-parts-lib, ... }:
+let
+  inherit (lib)
+    filterAttrs
+    mapAttrs
+    mkOption
+    optionalAttrs
+    types
+    ;
+  inherit (flake-parts-lib)
+    mkSubmoduleOptions
+    ;
+in
+{
+  options = {
+    flake = mkSubmoduleOptions {
+      flakePartsModules = mkOption {
+        type = types.lazyAttrsOf types.unspecified;
```

```suggestion
        type = types.lazyAttrsOf types.deferredModule;
```

This will add more detailed info about the individual definition(s), especially when the module is in its own file.

It's ok to keep the `apply` solution as well, because it creates a more stable name to use in `disabledModules`.

---

Should this change be applied to nixosModules and darwinModules as well? I just copied those.